R/plot.r

```diff
@@ -36,42 +45,55 @@
 #'   evaluation.
 #' @export
 #' @examples
-#' # Generate some sample data, then compute mean and standard deviation
-#' # in each group
+#' # Create a data frame with some sample data, then create a data frame
+#' # containing the mean value for each group in the sample data.
 #' set.seed(1)
-#' df <- data.frame(
-#'   gp = factor(rep(letters[1:3], each = 10)),
-#'   y = rnorm(30)
+#' 
+#' sample_df <- data.frame(
+#'   group = factor(rep(letters[1:3], each = 10)),
+#'   value = rnorm(30)
 #' )
-#' ds <- do.call(rbind, lapply(split(df, df$gp), function(d) {
-#'   data.frame(mean = mean(d$y), sd = sd(d$y), gp = d$gp)
-#' }))
-#'
-#' # The summary data frame ds is used to plot larger red points on top
-#' # of the raw data. Note that we don't need to supply `data` or `mapping`
-#' # in each layer because the defaults from ggplot() are used.
-#' ggplot(df, aes(gp, y)) +
+#' 
+#' group_means_df <- setNames(
+#'   aggregate(value ~ group, sample_df, mean),
+#'   c("group", "group_mean")
+#' )
+#' 
+#' # The following three code blocks create the same graphic, each using one
+#' # of the three patterns specified above. In each graphic, the group means
+#' # data frame is used to plot larger red points on top of the sample data.
+#' 
+#' # Pattern 1
+#' # Both the `data` and the `mapping` arguments are passed into the `ggplot()`
+#' # call. Note that we don't need to supply `mapping` or `data` arguments in
+#' # the first `geom_point()` layer because those arguments get passed along
+#' # from the `ggplot()` call.
+#' ggplot(data = sample_df, mapping = aes(x = group, y = value)) +
 #'   geom_point() +
-#'   geom_point(data = ds, aes(y = mean), colour = 'red', size = 3)
-#'
-#' # Same plot as above, declaring only the data frame in ggplot().
-#' # Note how the x and y aesthetics must now be declared in
-#' # each geom_point() layer.
-#' ggplot(df) +
-#'   geom_point(aes(gp, y)) +
-#'   geom_point(data = ds, aes(gp, mean), colour = 'red', size = 3)
-#'
-#' # Alternatively we can fully specify the plot in each layer. This
-#' # is not useful here, but can be more clear when working with complex
-#' # mult-dataset graphics
+#'   geom_point(mapping = aes(x = group, y = group_mean), data = group_means_df,
+#'              colour = 'red', size = 3
+#'     )
```

Comment:
I apologise profusely for being nit-picky about style here, but might I be able to convince you to indent the second point layer as follows according to the style guide's direction for [long lines](https://style.tidyverse.org/syntax.html#long-lines)? I think `mapping` and `data` can go on the same line because they're related. Similarly, `colour = 'red', size = 3` are both static aesthetics.

```r
  geom_point(
    mapping = ..., data = ...,
    colour = ..., size = ...
  )
```
This applies to the examples in pattern 2 and 3 as well.

Reply:
I did give this some thought.

While I'm all for style guides, I do think that an 80 character line length is too arbitrary of a criteria to be strictly enforced.

That said, I kept the line as it currently is for 2 reasons:
1. The line is 77 char without the leading "#' ", and exactly 80 with, 
2. I felt that the parallelism across the layers and across the 3 patterns made it much easier to see the differences in the patterns.

That is, I think this:
```
ggplot(data = sample_df) +
  geom_point(mapping = aes(x = group, y = value)) +
  geom_point(mapping = aes(x = group, y = group_mean), data = group_means_df,
             colour = 'red', size = 3
  )
```
is easier to read than this:
```
ggplot(data = sample_df) +
  geom_point(mapping = aes(x = group, y = value)) +
  geom_point(
    mapping = aes(x = group, y = group_mean), data = group_means_df,
    colour = 'red', size = 3
  )
```

And this:
```
ggplot() +
  geom_point(mapping = aes(x = group, y = value), data = sample_df) +
  geom_point(mapping = aes(x = group, y = group_mean), data = group_means_df,
             colour = 'red', size = 3
  )
```
is easier to read than this:
```
ggplot() +
  geom_point(
    mapping = aes(x = group, y = value), data = sample_df
  ) +
  geom_point(
    mapping = aes(x = group, y = group_mean), data = group_means_df,
    colour = 'red', size = 3
  )
```

If those don't sway you, though, I'll indent them as you suggest.

Reply:
I sympathise with your arguments: I too feel that a style guide is too restrictive at times and do agree that the parallelism has merit. The reason I suggested it is not because the style guide recommends it (though it is easy to point to), which has some value but shouldn't be followed blindly. Instead, I suggested it because it would be more internally consistent with various other examples in other doc pages, which I do feel is somewhat important. Perhaps we could get the opinion of a third person to decide for us. 

@thomasp85
TL;DR: do you have any opinion stylewise on the following two alternatives, or a suggestion of your own?

```r
#            Nice alignment of arguments
ggplot() + # V         V   V          V
  geom_point(mapping = aes(x = group, y = value), data = sample_df) +
  geom_point(mapping = aes(x = group, y = group_mean), data = group_means_df,
             colour = 'red', size = 3
  )
```

versus

```r
# Dogmatically consistent
ggplot() +
  geom_point(mapping = aes(x = group, y = value), data = sample_df) +
  geom_point(
    mapping = aes(x = group, y = group_mean), data = group_means_df,
    colour = 'red', size = 3
  )
```

Reply:
The argument for consistency in style across examples weigh the heaviest IMO